SpiralCompiler/Syntax/Lexer.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace SpiralCompiler.Syntax
+{
+    public class Lexer
+    {
+        private readonly string code;
+        private int index;
+
+        private bool IsEnd => index >= code.Length;
+
+        public Lexer(string code)
+        {
+            this.code = code;
+        }
+
+        public static List<Token> Lex(string code)
+        {
+            var lexer = new Lexer(code);
+            var tokenlist = new List<Token>();
+
+            while (true)
+            {
+                var token = lexer.Next();
+                tokenlist.Add(token);
+
+                if (token.Type == TokenType.EndOfFile) break;
+            }
+            return tokenlist;
+        }
+
+        public Token Next()
+        {
+        begin:
+            if (IsEnd) return new Token("", TokenType.EndOfFile, new Range(index, index));
+
+            var ch = Peek();
+            if (char.IsWhiteSpace(ch))
+            {
+                ++index;
+                goto begin;
+            }
+
+            // Comment
+            if (ch == '/' && Peek(1) == '/')
+            {
+                index += 2;
+                while (Peek(defaultChar: '\r') != '\r' && Peek(defaultChar: '\n') != '\n')
```

Comment:
Eh a bit funky but passable

Reply:
If you want to make this a bit nicer, you could just write an `IsNewline` helper.